### ISSUE TITLE
`velocity` option

### ISSUE THREAD

**Issue by luksak** [pisi/Reel#27]
Former "add an option to make a animation when the reel is initialized"

hi

i would like to create an animation as soon the reel is initialized, so that the user knows that he can rotate my product. after that i want no automatic rotation, but this feature already exists. also a nice feature would be to apply the break parameter, so that the product rotates fast in the beginning and the slows down.

if you are not able to implement this feature in the close future, could you give me a hint how i could create a workaround? for example trigger a mouse move event on the reel div as soon as it is initialized.

thanks in advance

lukas

**Comment by pisi** [comment]
Lukas,

applying the braking or inertia to ease the opening animation stop is a nice idea! It would very much improve it. I will consider it for 1.2 or some other future versions.

If you really want to workaround this, my advice would be to use the `velocity` data key which can be get/set with jQuery `.data()`. This is the calculated velocity of mouse movement right before mouse release, on which he brake is applied and cause Reel to stop. Try different values.

Please let me know what you come up with!

.pisi

**Comment by luksak** [comment]
Hey pisi

I tried doing the following without success:

```
$('#reel').data('velocity', 1);
$('#reel').reel();
```

Did I get you wrong? Where am I supposed to do this?

Thanks

Lukas

**Comment by luksak** [comment]
hey pisi

i managed to accomplish it with this code:

```
$('#reel').reel();
$('#reel').live('loaded', function(){
    $('#reel').data('velocity', 6.3);
});
```

reel is a really nice plugin! thank you for the great work!

lukas

**Comment by pisi** [comment]
Very well done, Lukas!

Looks like a `velocity` option would be a nice addition in 1.2. Would be nice to do just:

```
$('#reel').reel({ velocity: 6.3 });
```

**Comment by luksak** [comment]
true. thanks!

**Comment by reformatco** [comment]
Would be great to add a callback for when the animation ends. I have tried hooking onto the 'opening' or 'opening_tick' method but neither worked. So a callback like:

``` javascript
$('#reel').reel({
    onEntryEnd: function(){
        /* Do something */
    }
});
```

**Comment by pisi** [comment]
Hey Ben,
you are indeed right, there's an empty space for an event to be triggered when the opening animation finishes. Nice one! ;) Extracted it into #65 as it is kind of off-topic here.

**Comment by pisi** [comment]
So the `velocity` option is in merged into [development](/pisi/Reel/tree/development) branch. It will be shipped in 1.2 and it can be seen in action in the [test sampler](http://test.vostrel.cz/jquery.reel/test/sampler.html) in the "Clockwise Object Movie Sequence" sample.
